packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,230 @@
+  () => {
+    if (!inputElement.value) return;
+
+    const newValidityState = transformValidityStateToObject(inputElement.value.validity);
+    //  only update + emit the validity state when it changed
+    if (!validityState.value || !areObjectsFlatEqual(newValidityState, validityState.value)) {
+      validityState.value = newValidityState;
+      emit("validityChange", validityState.value);
+    }
+  },
+  { immediate: true },
+);
+</script>
+
+<template>
+  <label
+    class="onyx-switch"
+    :class="{ 'onyx-required-marker': props.required, 'onyx-optional-marker': !props.required }"
+  >
+    <input
+      ref="inputElement"
+      v-model="isChecked"
+      class="onyx-switch__input"
+      type="checkbox"
+      :aria-label="props.hideLabel ? props.label : undefined"
+      :disabled="props.disabled"
+      :required="props.required"
+    />
+    <span class="onyx-switch__container">
+      <span class="onyx-switch__icon">
+        <OnyxIcon :icon="isChecked ? checkSmall : xSmall" size="24px" />
+      </span>
+    </span>
+    <span v-if="!props.hideLabel" class="onyx-switch__label">{{ props.label }}</span>
+  </label>
+</template>
+
+<style lang="scss">
+.onyx-switch {
+  display: inline-flex;
+  align-items: center;
+  cursor: pointer;
+  margin: var(--onyx-spacing-2xs);
+  height: 1.5rem;
+
+  &__input {
+    // position: absolute is needed here in order to hide the native checkbox.
+    position: absolute;
+    opacity: 0;
+
+    &:checked + .onyx-switch__container {
+      background-color: var(--onyx-color-base-primary-500);
+
+      .onyx-switch__icon {
+        background-color: var(--onyx-color-themed-neutral-100);
+        transform: translateX(14px);
+        color: var(--onyx-color-text-icons-primary-intense);
+      }
+    }
+
+    &:checked:disabled + .onyx-switch__container {
+      background-color: var(--onyx-color-base-primary-200);
+
+      .onyx-switch__icon {
+        background-color: var(--onyx-color-base-background-blank);
+        color: var(--onyx-color-text-icons-primary-soft);
+      }
+    }
+
+    &:disabled + .onyx-switch__container {
+      background-color: var(--onyx-color-base-neutral-200);
+
+      .onyx-switch__icon {
+        background-color: var(--onyx-color-base-neutral-300);
+        color: var(--onyx-color-text-icons-neutral-inverted);
+      }
+    }
+
+    &:invalid + .onyx-switch__container {
+      background-color: var(--onyx-color-base-danger-200);
+      border: var(--onyx-1px-in-rem) solid var(--onyx-color-base-danger-500);
+      height: 1.375rem;
+      width: 2.25rem;
+
+      .onyx-switch__icon {
+        background-color: var(--onyx-color-base-danger-500);
+        color: var(--onyx-color-text-icons-neutral-inverted);
+      }
+    }
+
+    &:invalid:checked + .onyx-switch__container {
+      background-color: var(--onyx-color-base-danger-500);
+
+      .onyx-switch__icon {
+        background-color: var(--onyx-color-base-background-blank);
+        transform: translateX(12px);
```

Comment:
All the sizing are given in `rem`, so we should probably give a rem value here, too.